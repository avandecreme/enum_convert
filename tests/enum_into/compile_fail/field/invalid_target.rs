use enum_convert::EnumInto;

#[derive(EnumInto)]
#[enum_into(Target)]
enum Source {
    #[enum_into(Target::Stuff)]
    Data {
        #[enum_into(Target::Stuff.nonexistent)] // Invalid field name
        x: i32,
    },
}

enum Target {
    Stuff { x: i64 },
}

fn main() {}
