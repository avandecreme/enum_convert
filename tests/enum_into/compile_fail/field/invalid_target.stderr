error[E0559]: variant `Target::Stuff` has no field named `nonexistent`
 --> tests/enum_into/compile_fail/field/invalid_target.rs:8:29
  |
8 |         #[enum_into(Target::nonexistent)] // Invalid field name
  |                             ^^^^^^^^^^^ `Target::Stuff` does not have this field
  |
  = note: all struct fields are already assigned
